ncf_train/unittest_train.py

```diff
@@ -0,0 +1,155 @@
+import pandas
+import unittest
+import os
+import sys
+import json
+import shutil
+from train import add_timestamp_col, split_train_test, augment_data, drop_colums, get_data_attributes, fit_model, get_test_attributes, get_recall, save_pt_model
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+
+class test_train(unittest.TestCase):
+    @classmethod
+    def setUpClass(self):
+        # Create data for testing and define data parameters
+        df = pandas.DataFrame({ 
+            'user_id': [0,0,0,1,1,1,2,2,2,3,3,3,4,4,4],
+            'item_id': [0,0,0,1,1,1,2,2,2,3,3,3,4,4,4],
+            'weight':  [3,20,2,3,20,2,3,20,2,3,20,2,3,20,2]
+        })
+        self.test_data_len = len(set(df['user_id']))
+        self.train_data_len = len(df) - self.test_data_len
+        self.test_num_users = len(set(df['user_id']))
+        self.test_num_items = len(set(df['item_id']))
+        self.test_all_item_ids = list(set(df['item_id']))
+        self.epochs = 100
+        self.batch_size = 512
+        self.num_workers = 0
+
+        # Make a Unit-testing directory
+        self.unittest_dir = "unit_test_data"
+        self.local_dir = os.path.dirname(os.path.abspath(__file__))
+        self.data_path = os.path.join(self.local_dir, self.unittest_dir)
+        os.mkdir(self.data_path)
+
+        # Store the csv data in the unittesting directory
+        df.to_csv(self.data_path + '/' + 'data.csv', index=False)
+
+        # Create test data frame
+        self.ratings = add_timestamp_col(self.data_path, 'data.csv')
+        # Split data into train and test
+        self.train_ratings, self.test_ratings = split_train_test(self.ratings)
+        # Augment train data
```

Comment:
Wondering if the code under test provide a function to train the model as per these init lines?